modules/data_integrity_flag/php/data_integrity_flag.class.inc

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * This file contains the Data_Integrity_Flag class
+ *
+ * PHP Version 5
+ *
+ * @category Module
+ * @package  Data_Integrity_Flag
+ * @author   Alex Ilea <ailea.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/CCNA/
+ */
+namespace LORIS\data_integrity_flag;
+
+/**
+ * Main class for data_integrity_flag module
+ * corresponding to /data_integrity_flag/ URL
+ *
+ * PHP Version 5
+ *
+ * @category Module
+ * @package  Data_Integrity_Flag
+ * @author   Alex Ilea <ailea.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/CCNA/
+ */
+class Data_Integrity_Flag extends \NDB_Menu_Filter
+{
+    public $AjaxModule = true;
+
+    /**
+     * Check user permissions
+     *
+     * @return bool
+     * @throws ConfigurationException
+     */
+    function _hasAccess()
+    {
+        //create user object
+        $user =& \User::singleton();
+        return $user->hasPermission('data_integrity_flag');
+    }
+
+    /**
+     * Displays the menu page
+     *
+     * @return void
+     * @access public
+     */
+    function display()
+    {
+        if ($_SERVER['REQUEST_METHOD'] === "GET") {
+            return parent::display();
+        }
+
+        if ($_SERVER['REQUEST_METHOD'] === "POST") {
+            $stdin = file_get_contents('php://input');
+            $req   = json_decode($stdin, true);
+            $this->_insertDataFlag($req);
+        }
+    }
+
+    /**
+     * Inserts form data in the database
+     *
+     * @param object $req HTTP request data
+     *
+     * @return void
+     * @access private
+     */
+    function _insertDataFlag($req)
+    {
+        $user =& \User::singleton();
+        $db   =& \Database::singleton();
+
+        if (!isset($req['instrument'])
+            || !isset($req['visitLabel'])
+            || !isset($req['comment'])
+            || !isset($req['date'])
+            || !isset($req['flagStatus'])
+        ) {
+            return "Error";
+        }
+
+        try {
```

Comment:
This looks like a hack and I'm not sure why it's doing it. If the update or insert statements are throwing exceptions, the root cause should be fixed instead of silently catching and discarding the errors..